internal/pkg/manifest/validate.go

```diff
@@ -87,13 +87,16 @@ func (l LoadBalancedWebService) validate() error {
 	}); err != nil {
 		return fmt.Errorf("validate HTTP load balancer target: %w", err)
 	}
-	if err = validateTargetContainer(validateTargetContainerOpts{
-		mainContainerName: aws.StringValue(l.Name),
-		mainContainerPort: l.ImageConfig.Port,
-		targetContainer:   l.NLBConfig.TargetContainer,
-		sidecarConfig:     l.Sidecars,
-	}); err != nil {
-		return fmt.Errorf("validate network load balancer target: %w", err)
+	nlbRoutingRules := l.NLBConfig.NLBRoutingRules()
+	for _, rule := range nlbRoutingRules {
+		if err = validateTargetContainer(validateTargetContainerOpts{
+			mainContainerName: aws.StringValue(l.Name),
+			mainContainerPort: l.ImageConfig.Port,
+			targetContainer:   rule.TargetContainer,
+			sidecarConfig:     l.Sidecars,
+		}); err != nil {
+			return fmt.Errorf("validate network load balancer target: %w", err)
+		}
```

Comment:
what do you think of including the index for which listener's target container we are validating?

For example:
```go
return fmt.Errorf("validate target for nlb.listener[%i]: %w", i, err)
```